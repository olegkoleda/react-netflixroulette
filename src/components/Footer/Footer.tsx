import React from "react";
import styled from "styled-components";
import { Box } from "atomic-layout";
import Logo from "../Logo";

const StyledFooter = styled.div`
  height: 100%;
  background: ${(props) => props.theme.colour.darkgray};
  text-align: center;
`;

export const Footer = () => (
  <Box as={StyledFooter} flex justifyContent="center" alignItems="center">
    <Logo />
  </Box>
);
