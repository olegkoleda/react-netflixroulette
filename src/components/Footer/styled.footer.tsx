import styled from "styled-components";

export const StyledFooter = styled.footer`
  height: 100%;
  background: ${(props) => props.theme.color.darkgray};
  text-align: center;
`;
