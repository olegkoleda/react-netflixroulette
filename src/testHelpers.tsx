import React from "react";
import renderer from "react-test-renderer";
import { theme } from "./styles/Theme";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { initialAppState } from "./store/reducers/rootReducer";

const middleWares = [thunk];
const mockStore = configureMockStore(middleWares);
const store = mockStore(initialAppState);

export function renderWithTheme(component: React.ReactNode) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export const reduxWrapper = (Component: typeof React.Component | React.FC, data = {}) => (
  <Provider store={store}>
    <Router>
      <Component {...data}/>
    </Router>
  </Provider>
);
