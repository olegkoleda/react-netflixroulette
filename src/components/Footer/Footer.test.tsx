import React from "react";
import Footer from "./Footer";
import { renderWithTheme } from "../../testHelpers";

test("FooterSection renders correctly", () => {
  const component = renderWithTheme(<Footer />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
