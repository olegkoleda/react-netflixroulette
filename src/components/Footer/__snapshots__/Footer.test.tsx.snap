// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterSection renders correctly 1`] = `
<footer
  className="styledfooter__StyledFooter-vo699h-0 cZSprY sc-AxjAm bqPLoD"
  height="4.5rem"
>
  <span>
    <span
      className="styledlogo__StyledLogo-sc-1suyno6-0 dKPWah"
      fontWeight={800}
    >
      netflix
    </span>
    <span
      className="styledlogo__StyledLogo-sc-1suyno6-0 eKILPN"
      fontWeight={400}
    >
      roulette
    </span>
  </span>
</footer>
`;
